R/get_header_groups.R

```diff
@@ -0,0 +1,204 @@
+#' Get column groups
+#'
+#' This is an internal function that:
+#'          1. Identifies which cells are likely to be headers
+#'          2. groups them according to their indenting, bold and italic formatting
+#'          3. Specifies the unpivotr function specifying the direction of the header w.r.t. table data
+#' Behead multiple header groups 
+#' @description
+#' Beheads multiple headers defined according to expressions in .groupings. 
+#' @param sheet data frame created by xlsx_cells
+#' @param direction  a string indicating which type of headers are to be labelled. Options include compass direction or up/down/left/right. 
+#' @param value_ref  reference to where data cells are located. 
+#' @param table_data datacell dataframe.
+#' @param formats  format object created by tidyxl. 
+#' @param .hook_if expression determining whether direction is hooked.
+#' @param .hook_if_rev expression determining whether direction is reverse hooked.
+#' @param .groupings expressions representing how header cells are differentiated. Most naturally works with fmt_* functions. 
+#' @param default_col_header_direction Indicates which direction is given to col headers by default. Only need if "NNW" is required, rather than "N". 
+#' @param header_fill deals with merged cells. Fills in neighbouring cells if they have the same "local_format_id", "style" or are within "borders".
+#' @param min_header_index min header index
+
+get_header_groups <- function(sheet, direction, value_ref, formats,
+                              .groupings = groupings(fmt_alignment_indent),
+                              .hook_if = hook(ones),
+                              .hook_if_rev = hook(ones),
+                              header_fill = "local_format_id",
+                              default_col_header_direction = default_col_header_direction_temp,
+                              table_data = tabledata,
+                              min_header_index = min_header_index_temp) {
```

Comment:
Were some of these default values left here by mistake after copy/pasting?

Reply:
Hmm. I'll have to revisit this. Looking at this now,  groupings and hooks could have more neutral defaults.

Reply:
On second thought, I think these are ok. I've only changed `.groupings` to take `groupings(ones)` in 7de8a7bb8296c49104dde259c31d58374f4fe40c. ones is just a function that returns ones, so it leads to no grouping at all. 

I may be missing something here, so I'm happy to discuss.